core/modules/filter/js/filter.js

```diff
@@ -333,6 +329,87 @@ Backdrop.behaviors.editorImageLibrary = {
         var $submit = $form.find('.form-actions input[type=submit]:first');
         $submit.trigger('mousedown').trigger('click').trigger('mouseup');
       });
+
+    // Lock image aspect ratio.
+    var naturalDimensions = {
+      width: null,
+      height: null
+    };
+    // When editing a previously added image or upload a new one.
+    var existingFile = $('.filter-format-editor-image-form .form-managed-file a').attr('href');
+    if (typeof existingFile !== 'undefined') {
+      let img = new Image();
+      img.onload = function() {
+        naturalDimensions.width = this.width;
+        naturalDimensions.height = this.height;
+        if (!$('.filter-format-editor-image-form [name="attributes[width]').val().length) {
+          Backdrop.behaviors.editorImageLibrary.imageDimensionsSet(naturalDimensions);
+        }
+        Backdrop.behaviors.editorImageLibrary.syncAspectRatio(naturalDimensions);
+      }
+      img.onerror = function() {
+        naturalDimensions.width = null;
+        naturalDimensions.height = null;
+      }
+      img.src = existingFile;
+    }
+    else if ($('.filter-format-editor-image-form [name="attributes[width]').length) {
+      // After an image has been removed via button, and the managed form item
+      // reloads, reset width and height.
+      if ($('.filter-format-editor-image-form [name="attributes[width]').val().length) {
+        naturalDimensions.width = null;
+        naturalDimensions.height = null;
+        Backdrop.behaviors.editorImageLibrary.imageDimensionsEmpty();
+      }
+    }
+
+    // Selecting an image from library updates width and height values.
+    $('.filter-format-editor-image-form [name="attributes[src]"]').once().on('change', function() {
```

Comment:
Oh, one more thing. I don't have a ton of experience with `once()`, but I believe it's better to provide an id in `once()`, otherwise the default id `once` will be used. Not sure about this, but it may create problems if there are more than one `once()` using the default id in the document. This id can be anything (it's not a DOM element id, but rather an internal id to keep track of methods that have been executed). 

All uses of `once` in Backdrop provide an id

Reply:
Good idea. :+1: